magstock/__init__.py

```diff
@@ -0,0 +1,75 @@
+from uber.common import *
+from magstock._version import __version__
+
+config = parse_config(__file__)
+static_overrides(join(config['module_root'], 'static'))
+template_overrides(join(config['module_root'], 'templates'))
+
+
+@Config.mixin
+class ExtraConfig:
+    @property
+    def FOOD_COUNT(self):
+        with Session() as session:
+            return session.query(Attendee).filter_by(purchased_food=True).count()
+
+    @property
+    def PREREG_DONATION_OPTS(self):
+        """
+        We're overriding this so that we can cut off Supporter registrations
+        after we've sold a capped number and so that we can cut off shirt sales
+        after the supporter deadline (MAGStock doesn't assume inventory risk on
+        tshirts so they only sell them to people who preorder and do not sell
+        any on-site).
+        """
+        if self.AFTER_SUPPORTER_DEADLINE:
+            return [(amt, desc) for amt, desc in self.DONATION_TIER_OPTS if amt < self.SHIRT_LEVEL]
+        elif not self.SUPPORTER_AVAILABLE:
+            return [(amt, desc) for amt, desc in self.DONATION_TIER_OPTS if amt < self.SUPPORTER_LEVEL]
+        else:
+            return self.DONATION_TIER_OPTS
+
+
```

Comment:
This is how we add and override things on our database models.  Here we're doing several things:

1) Adding new columns, which we do simply by declaring them as we would in the core Uber plugin.  We can combine this with the ``regextras.html`` template overriding mentioned above to add new form fields which are tracked in the database.

2) We added a new action which happens immediately before an object is saved to the database.  In this case we assign a "Roughing It" ribbon to anyone who selects the "Primitive" campsite type.

3) MAGStock sells food with its registrations for an extra $35.  This is NOT a kick-in level, which means that we need to do two things in order to have this recognized as an extra thing for sale:
* Create a new ``@cost_property`` which returns the amount extra that someone is paying (either $35 or $0) based on whether or not they've checked the box which indicates they want food.
* Override the ``addons`` property, which is used to list things which people buy in confirmation emails and such.

There are theoretically other things we could do here, including but not limited to:
* removing or changing existing columns which are defined in the core Uber plugin if we don't want to store them or if we want to make them optional or something
* changing the behavior of existing properties and methods
* adding new pre-delete actions
* probably other stuff I'm forgetting :P (but seriously, if there are other things we want to do, we can make an API for them if they don't exist already)